www/class/centreonDowntime.class.php

```diff
@@ -494,6 +494,36 @@ public function duplicate($ids, $nb)
         } else {
             $ids = $this->normalizeArray($ids);
         }
+        $query = 'SELECT COUNT(*) as nb FROM downtime WHERE dt_name = :dt_name';
```

Comment:
It's a good idea to use secure sql request 👍 

juste some sugesstion : 
- using transaction would be perfect : we could avoid orpelan records.
- The method "duplicate" is getting too big : maybe you could refactor it. For exemple 
````               while ($i <= $nb[$id]) {
                        if (! $this->downTimeExists($dt_name)) {
                            $this->createDowntime($params);
                            $this->createPeriod($params);
                            $this->createHostRelation($params);
                            $this->createServiceRelation($params);
                            $this->createServicegroupRelation($params);
                        }
                        $i++;
                    }
                    ```

Reply:
@TamazC done :)